### ISSUE TITLE
`comparison-overlap` not triggered for pathlib.Path

### ISSUE THREAD

**Issue by NMertsch** [python/mypy#16041]
`--strict-comparison` seems to assume that `pathlib.Path` objects can be equal to anything.

**To Reproduce** ([playground](https://mypy-play.net/?mypy=latest&python=3.11&flags=strict-equality&gist=f87d542ff65df4ef507e73244fe1e538))

```python
# run with --strict-equality
from pathlib import Path

class A: pass

'x' == 1  # comparison-overlap: this is always false
A() == 1  # comparison-overlap: this is always false
Path('x') == 1  # fine for mypy
```

**Expected Behavior**
Both lines should trigger 'non-overlapping equality check'

**Actual Behavior**
Only `'x' == 1` triggers 'non-overlapping equality check'

**Comment by hauntsaninja** [comment]
This is because of https://github.com/python/typeshed/pull/7231/files#diff-b5aebc9806d203cfe171e2215b0b4f3106395fc59687382eb9276d7d8976bffd , transferring to typeshed
